pgmpy/inference/base.py

```diff
@@ -80,3 +80,40 @@ def __init__(self, model):
             for factor in model.get_factors():
                 for var in factor.variables:
                     self.factors[var].append(factor)
+
+
+class ApproxInference:
+    """
+    Base class for the approximate inference algorithm.
+    Parameters
+    ----------
+    model: pgmpy.models.MarkovModel
+    We need a pairwise MarkovModel for Approx. inference.
+    """
+    def __init__(self, model):
+
+        # List of the variables in the present pairwise Markov Model
+        self.variables = []
+
+        # List of the cardinalities of the above variables in that order
+        self.cardinality = []
+
+        # List of all the factors that are present in the current model
+        self.factors = []
```

Comment:
Same. Don't initialize variables for storing the factors in inference class. You can get them directly from the ```model.get_factors()```. If you want to store factors corresponding to each variable (in form of a dictionary), then you can inherit from the ```Inference``` class; it computes this during initialization itself.